public/app/plugins/datasource/influxdb/components/RawInfluxQLEditor.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC, useState, useEffect } from 'react';
+import { TextArea, InlineFormLabel, Input, Select, HorizontalGroup } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { ResultFormat, InfluxQuery } from '../types';
+
+const RESULT_FORMATS: Array<SelectableValue<ResultFormat>> = [
+  { label: 'Time series', value: 'time_series' },
+  { label: 'Table', value: 'table' },
+  { label: 'Logs', value: 'logs' },
+];
+
+const DEFAULT_RESULT_FORMAT: ResultFormat = 'time_series';
+
+type Props = {
+  query: InfluxQuery;
+  onQueryChange: (query: InfluxQuery) => void;
+  onRunQuery: () => void;
+};
+
+// we handle 3 fields: "query", "alias", "resultFormat"
+// "resultFormat" changes are applied immediately
+// "query" and "alias" changes only happen on onblur
+export const RawInfluxQLEditor: FC<Props> = ({ query, onQueryChange, onRunQuery }) => {
+  const [currentQuery, setCurrentQuery] = useState(query.query);
+  const [currentAlias, setCurrentAlias] = useState(query.alias);
+
+  useEffect(() => {
+    // if the value changes from the outside, we accept it
+    if (currentQuery !== query.query) {
+      setCurrentQuery(query.query);
+    }
+  }, [query.query]);
+
+  useEffect(() => {
+    // if the value changes from the outside, we accept it
+    if (currentAlias !== query.alias) {
+      setCurrentAlias(query.alias);
+    }
+  }, [query.alias]);
+
+  const applyDelayedChangesAndRunQuery = () => {
+    onQueryChange({
+      ...query,
+      query: currentQuery,
+      alias: currentAlias,
+    });
+    onRunQuery();
+  };
+
+  return (
+    <div>
+      <TextArea
+        rows={3}
+        spellCheck={false}
+        placeholder="InfluxDB Query"
+        onBlur={applyDelayedChangesAndRunQuery}
+        onChange={(e) => {
+          setCurrentQuery(e.currentTarget.value);
+        }}
+        value={currentQuery ?? ''}
+      />
+      <HorizontalGroup>
+        <InlineFormLabel>Format as</InlineFormLabel>
+        <Select
+          onChange={(e) => {
+            onQueryChange({ ...query, resultFormat: e.value });
+            onRunQuery();
+          }}
+          value={query.resultFormat ?? DEFAULT_RESULT_FORMAT}
+          options={RESULT_FORMATS}
+        />
+        <InlineFormLabel>Alias by</InlineFormLabel>
+        <Input
+          type="text"
+          spellCheck={false}
+          placeholder="Naming pattern"
+          onBlur={applyDelayedChangesAndRunQuery}
```

Comment:
@ivana do you know how we deal with running things on blur in Explore? I always forget but I assume we should be checking some sort of context to know if we are in dashboard or explore? Or is there something else?

@gabor probably not required to change here as this is just refactor but there is a difference that in dashboard we want to run queries on blur but in Explore we want to be more explicit and run either on shift + enter or on button click. And we always have issues with that as each query editor needs to be aware of that.

Reply:
@aocenas thanks for the info. 
i checked the prometheus datasource and maybe it's done by setting up both `.setQueryEditor()` and `.setExploreMetricsQueryField()`, and having different react-components for the dashboard/explore this way (they both wrap the same component), one of them handles shift-click etc.
unfortunately the "main" query-editor in this pull-request is still in angular, and i cannot find any way to distinguish between the two cases there, and also the current version of influx just does the runQuery on onBlur, so there's no regression.
so i think we just let it be like it is for the refactor, and i will make sure it works the expected way when the "main" part of the query-editor switches to react; basically, what you said :-)